openmdao/core/problem.py

```diff
@@ -409,6 +409,16 @@ def setup(self, check=True, out_stream=sys.stdout):
         # a single source.
         connections = self._setup_connections(params_dict, unknowns_dict)
 
+        # Allow the user to omit the size of a parameter and pull the size
+        # and shape from the connection source.
+        for tgt, src in iteritems(connections):
+            tmeta = params_dict[tgt]
+            if not tmeta.get('pass_by_obj') and tmeta['shape'] == ():
+                smeta = unknowns_dict[src[0]]
+                tmeta['shape'] = smeta['shape']
+                tmeta['size'] = smeta['size']
+                tmeta['val'] = smeta['val']
```

Comment:
What happens if there are src_indices in the connection?   size/shape would then be implied but it seems like copying the meta from the source would be incorrect in that case?  Is this handled somewhere else?